libs/area_reporting.py

```diff
@@ -0,0 +1,72 @@
+import os
+import time
+import logging
+import csv
+from datetime import date
+from collections import deque
+from .utils.mailing import MailService
+from .notifications.slack_notifications import SlackService
+
+logger = logging.getLogger(__name__)
+
+
+class AreaReporting:
+
+    def __init__(self, config, area):
+        self.processing_alerts = False
+        self.config = config
+        self.area = area
+
+        self.occupancy_sleep_time_interval = float(self.config.get_section_dict("App")["OccupancyAlertsTimeout"])
+        self.log_dir = self.config.get_section_dict("Logger")["LogDirectory"]
+        self.idle_time = float(self.config.get_section_dict('Logger')['TimeInterval']) + 0.1
+        self.area_id = self.area['id']
+        self.area_name = self.area['name']
+        self.occupancy_threshold = self.area['occupancy_threshold']
+        self.emails = self.area['emails']
+        self.should_send_email_notifications = self.area['should_send_email_notifications']
+        self.should_send_slack_notifications = self.area['should_send_slack_notifications']
+        self.cameras = [camera for camera in self.config.get_video_sources() if camera['id'] in self.area['cameras']]
+        for camera in self.cameras:
+            camera['file_path'] = os.path.join(self.log_dir, camera['id'], "objects_log")
+            camera['last_processed_time'] = time.time()
+
+        self.mail_service = MailService(config)
+        self.slack_service = SlackService(config)
+
+    def process_area(self):
+        # Sleep for a while so cameras start processing
+        time.sleep(30)
+
+        self.processing_alerts = True
+        logger.info(f'Enabled processing alerts for - {self.area_id}: {self.area_name} with {len(self.cameras)} cameras')
+        while self.cameras and self.processing_alerts:
+            camera_file_paths = [os.path.join(camera['file_path'], str(date.today()) + ".csv") for camera in self.cameras]
+            if not all(list(map(os.path.isfile, camera_file_paths))):
+                # Wait before csv for this day are created
+                logger.info(f'Area reporting on - {self.area_id}: {self.area_name} is waiting for reports to be created')
+                time.sleep(5)
+
+            occupancy = 0
+            for camera in self.cameras:
+                with open(os.path.join(camera['file_path'], str(date.today()) + ".csv"), 'r') as log:
+                    last_log = deque(csv.DictReader(log), 1)[0]
+                    occupancy += int(last_log['DetectedObjects'])
```

Comment:
I think that we should also check the `last_log` time.
If for some reason a camera stops to register the information in the csv log, the occupancy variable will summarize objects from different hours.

Reply:
Yeah, I thought about doing this but I wasn't sure if it was worth it. I fought it by adding the timer so the cameras start processing before this thread, but I don't know exactly how we should cope with a camera that stops logging data (Whether should we ignore it for the occupancy of the area or do something as take the median of the last 10 readings)